github/github.go

```diff
@@ -564,14 +576,18 @@ func (e *Error) Error() string {
 }
 
 // CheckResponse checks the API response for errors, and returns them if
-// present.  A response is considered an error if it has a status code outside
-// the 200 range.  API error responses are expected to have either no response
+// present. A response is considered an error if it has a status code outside
+// the 200 range or equal to 202 Accepted.
+// API error responses are expected to have either no response
 // body, or a JSON response body that maps to ErrorResponse.  Any other
 // response body will be silently ignored.
 //
 // The error type will be *RateLimitError for rate limit exceeded errors,
 // and *TwoFactorAuthError for two-factor authentication errors.
```

Comment:
Let's mention `AcceptedError` here.

```
// The error type will be *RateLimitError for rate limit exceeded errors,
// AcceptedError for 202 Accepted status codes,
// and *TwoFactorAuthError for two-factor authentication errors.
```

This makes me realize that there's actually a good reason to revert back to using a pointer to `*AcceptedError` rather than value: for consistency of type assertion code to other error types. Currently, it'd be:

```Go
switch err.(type) {
case *github.RateLimitError:
	// handle rate limit exceeded error
case *github.TwoFactorAuthError:
	// handle two-factor authentication error
case github.AcceptedError: // inconsistent
	// handle accepted error
}
```

If we revert to using a pointer to `*AcceptedError`, it becomes consistent.

```Go
switch err.(type) {
case *github.RateLimitError:
	// handle rate limit exceeded error
case *github.TwoFactorAuthError:
	// handle two-factor authentication error
case *github.AcceptedError: // consistent
	// handle accepted error
}
```

I think that's a good argument to use pointer here too. Do you agree?